src/solver/NSSolver.cpp

```diff
@@ -6,40 +6,38 @@
 
 #include "NSSolver.h"
 #include "../pressure/VCycleMG.h"
-#include "../utility/Parameters.h"
 #include "../Domain.h"
 #include "SolverSelection.h"
 #include "../boundary/BoundaryData.h"
 
-NSSolver::NSSolver(FieldController *field_controller) {
+NSSolver::NSSolver(Settings const &settings, FieldController *field_controller) :
+        m_settings(settings) {
 #ifndef BENCHMARKING
-    m_logger = Utility::create_logger(typeid(this).name());
+    m_logger = Utility::create_logger(m_settings, typeid(this).name());
 #endif
     m_field_controller = field_controller;
 
-    auto params = Parameters::getInstance();
-
     //Advection of velocity
-    std::string advectionType = params->get("solver/advection/type");
-    SolverSelection::SetAdvectionSolver(&adv_vel, advectionType);
+    std::string advectionType = m_settings.get("solver/advection/type");
+    SolverSelection::SetAdvectionSolver(m_settings, &adv_vel, advectionType);
 
     //Diffusion of velocity
-    std::string diffusionType = params->get("solver/diffusion/type");
-    SolverSelection::SetDiffusionSolver(&dif_vel, diffusionType);
+    std::string diffusionType = m_settings.get("solver/diffusion/type");
+    SolverSelection::SetDiffusionSolver(m_settings, &dif_vel, diffusionType);
 
-    m_nu = params->get_real("physical_parameters/nu");
+    m_nu = m_settings.get_real("physical_parameters/nu");
```

Comment:
remove member variable `m_nu` instead access value through Settings

Reply:
done in 0d1d248